pkg/cloud/scope/machine.go

```diff
@@ -183,6 +184,21 @@ func (m *MachineScope) SetAddresses(addrs []corev1.NodeAddress) {
 	m.AWSMachine.Status.Addresses = addrs
 }
 
+// GetBootstrapData returns the bootstrap data from the secret in the Machine's bootstrap.dataSecretName.
+func (m *MachineScope) GetBootstrapData() (string, error) {
+	if m.Machine.Spec.Bootstrap.DataSecretName == nil {
+		return "", errors.New("error retrieving bootstrap data: linked Machine's bootstrap.dataSecretName is nil")
```

Comment:
This error should probably include the machine's ns & name?

Reply:
I added the AWSMachine namespace and name from the caller, so all the errors will get it